tests/cupy_tests/core_tests/test_userkernel.py

```diff
@@ -67,6 +68,132 @@ def test_numpy_scalar(self, dtype):
         testing.assert_array_equal(out1, expected)
 
 
+class TestElementwiseKernelSize(unittest.TestCase):
+    # Tests to check whether size argument raises ValueError correctly
+    # depending on the raw specifiers of a user kernel.
+
+    def setUp(self):
+        self.arr1 = cupy.array([1, 2], dtype='float32')
+        self.arr2 = cupy.array([3, 4], dtype='float32')
+
+    def raises_size_not_allowed(self):
+        return pytest.raises(ValueError, match=r'^Specified \'size\' can')
+
+    def raises_size_required(self):
+        return pytest.raises(ValueError, match=r'^Loop size is undecided\.')
+
+    def create_kernel(self, input_raw, output_raw):
+        # Creates a no-op kernel with given parameter specification.
+        # input_raw and output_raw are tuples of True/False whose
+        # corresponding parameter will be designated as 'raw' if True.
+        input_types = (
+            ', '.join([
+                '{}float32 x{}'.format(
+                    ('raw ' if raw else ''), i)
+                for i, raw in enumerate(input_raw)]))
+        output_types = (
+            ', '.join([
+                '{}float32 y{}'.format(
+                    ('raw ' if raw else ''), i)
+                for i, raw in enumerate(output_raw)]))
+        return cupy.ElementwiseKernel(input_types, output_types, '', 'kernel')
+
+    def test_all_raws(self):
+        # Input arrays are all raw -> size required
+        kernel1 = self.create_kernel((True, True), (False,))
+        kernel1(self.arr1, self.arr2, size=2)
+        with self.raises_size_required():
+            kernel1(self.arr1, self.arr2)
+        kernel2 = self.create_kernel((True, True), (True,))
+        kernel2(self.arr1, self.arr2, size=2)
+        with self.raises_size_required():
+            kernel2(self.arr1, self.arr2)
+
+    def test_all_nonraws(self):
+        # All arrays are not raw -> size not allowed
+        kernel1 = self.create_kernel((False, False), (False,))
+        with self.raises_size_not_allowed():
+            kernel1(self.arr1, self.arr2, size=2)
+        kernel2 = self.create_kernel((False, False), (True,))
```

Comment:
Could these multiple kernel calls in all the tests be parametrized?

Reply:
Parameterization would also be pretty complicated.
It can be surely done, but I'm not sure it's worth the effort.

Reply:
I agree with your answer.